api/provider/aws/service_list.go

```diff
@@ -9,102 +9,76 @@ import (
 )
 
 func (s *ServiceProvider) List() ([]models.ServiceSummary, error) {
-	clusterArns, err := s.listClusters()
+	clusterNames, err := listClusterNames(s.AWS.ECS, s.Config.Instance())
 	if err != nil {
 		return nil, err
 	}
 
-	clusterServices := map[*string][]*string{}
-
-	for _, clusterArn := range clusterArns {
-		clusterName := strings.Split(aws.StringValue(clusterArn), ":")[5]
-		clusterName = strings.Replace(clusterName, "cluster/", "", -1)
-
-		if !strings.HasPrefix(clusterName, s.Config.Instance()) {
-			continue
-		}
-
-		serviceArns, err := s.listServices(clusterArn)
-		if err != nil {
-			return nil, err
-		}
+	serviceARNs, err := s.listServiceARNsForClusterNames(clusterNames)
+	if err != nil {
+		return nil, err
+	}
 
-		clusterServices[clusterArn] = serviceArns
+	serviceIDs := make([]string, len(serviceARNs))
+	for i, serviceARN := range serviceARNs {
+		fqServiceID := strings.Split(serviceARN, "/")[1]
```

Comment:
[nit] can you throw a comment right here showing a service ARN looks like?

Reply:
:white_check_mark: